### PR TITLE
(maint) Update changelog and metadata for release

### PR DESCRIPTION
**Pull request by jpogran**
Changelog updated with all changes since 2.0.2 release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+- Address json pure version problems ([MODULES-3640](https://tickets.puppetlabs.com/browse/MODULES-3640))
+- Capture exit codes when executing external scripts ([MODULES-3399](https://tickets.puppetlabs.com/browse/MODULES-3399))
+- Update appveyor and travis to use Ruby 2.3 ([MODULES-3775](https://tickets.puppetlabs.com/browse/MODULES-3775))
+- Add ability to set current working directory ([MODULES-3565](https://tickets.puppetlabs.com/browse/MODULES-3565))
+- Respect user specified timeout ([MODULES-3709](https://tickets.puppetlabs.com/browse/MODULES-3709))
+- Improve handling of user code exceptions ([MODULES-3443](https://tickets.puppetlabs.com/browse/MODULES-3443))
+- Output line and stacktrace of user code exception ([MODULES-3839](https://tickets.puppetlabs.com/browse/MODULES-3839))
+- Improve resliance to failure of powershell host ([MODULES-3875](https://tickets.puppetlabs.com/browse/MODULES-3875))
+- Fix race condition in threading with PowerShell host ([MODULES-3144](https://tickets.puppetlabs.com/browse/MODULES-3144))
+- Modify tests to detct differences in PowerShell error text ([MODULES-3442](https://tickets.puppetlabs.com/browse/MODULES-3442))
```
Comment (ferventcoder) [inline CHANGELOG.md]: typo

Comment (jpogran) [inline CHANGELOG.md]: Adjusted

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,24 @@
+##2016-10-03 - Supported Release 2.0.3
+###Summary
+
+Small release with bugs fixes and another speed improvement.
+
+###Bug Fixes
+- Miscellanous fixes which improve reliability
+- Address json pure version problems ([MODULES-3640](https://tickets.puppetlabs.com/browse/MODULES-3640))
+- Capture exit codes when executing external scripts ([MODULES-3399](https://tickets.puppetlabs.com/browse/MODULES-3399))
+- Update appveyor and travis to use Ruby 2.3 ([MODULES-3775](https://tickets.puppetlabs.com/browse/MODULES-3775))
```
Comment (ferventcoder) [inline CHANGELOG.md]: Actually remove this one, not important to the changelog (present functionality of the module itself)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,24 @@
...
+- Add ability to set current working directory ([MODULES-3565](https://tickets.puppetlabs.com/browse/MODULES-3565))
+- Respect user specified timeout ([MODULES-3709](https://tickets.puppetlabs.com/browse/MODULES-3709))
+- Improve handling of user code exceptions ([MODULES-3443](https://tickets.puppetlabs.com/browse/MODULES-3443))
+- Output line and stacktrace of user code exception ([MODULES-3839](https://tickets.puppetlabs.com/browse/MODULES-3839))
+- Improve resliance to failure of powershell host ([MODULES-3875](https://tickets.puppetlabs.com/browse/MODULES-3875))
```
Comment (ferventcoder) [inline CHANGELOG.md]: PowerShell

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+- Fix race condition in threading with PowerShell host ([MODULES-3144](https://tickets.puppetlabs.com/browse/MODULES-3144))
+- Modify tests to detect differences in PowerShell error text ([MODULES-3442](https://tickets.puppetlabs.com/browse/MODULES-3442))
+
+###Documentation updates
+- Document how to handle exit codes ([MODULES-3588](https://tickets.puppetlabs.com/browse/MODULES-3588))
```
Comment (ferventcoder) [inline CHANGELOG.md]: Not sure on this - I typically don't mention documentation changes. It can't hurt though.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,24 @@
+##2016-10-03 - Supported Release 2.0.3
+###Summary
+
+Small release with bugs fixes and another speed improvement.
+
+###Bug Fixes
+- Miscellanous fixes which improve reliability
+- Address json pure version problems ([MODULES-3640](https://tickets.puppetlabs.com/browse/MODULES-3640))
```
Comment (ferventcoder) [inline CHANGELOG.md]: Remove this one as well. Not something folks using the module tend to care about.

File: CHANGELOG.md (hunk 4 above)
Comment (jpogran) [inline CHANGELOG.md]: Was a big deal at the time...so I thought to call it out

Comment (ferventcoder) [inline CHANGELOG.md]: It's fine to leave this one.

**Pull request merged**